panic in processReceivedChunks

an unexpected panic while running `swarm-network-rewrite`:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0xbba2e9]

goroutine 45 [running]:
github.com/ethereum/go-ethereum/swarm/network/stream.(*Delivery).processReceivedChunks(0xc420a0b4a0)
	/home/e/go/src/github.com/ethereum/go-ethereum/swarm/network/stream/delivery.go:212 +0x119
created by github.com/ethereum/go-ethereum/swarm/network/stream.NewDelivery
	/home/e/go/src/github.com/ethereum/go-ethereum/swarm/network/stream/delivery.go:60 +0xbe
```

--- a/swarm/network/stream/delivery.go
+++ b/swarm/network/stream/delivery.go
@@ -17,7 +17,6 @@
 package stream
 
 import (
-	"bytes"
 	"errors"
 	"fmt"
 	"time"
@@ -210,9 +209,6 @@ R:
 		// this should be has locally
 		chunk, err := d.db.Get(req.Key)
 		if err == nil {
-			if !bytes.Equal(chunk.Key, req.Key) {
-				panic(fmt.Errorf("processReceivedChunks: chunk key %s != req key %s (peer %s)", chunk.Key.Hex(), req.Key.Hex(), req.peer.ID()))
-			}
 			continue R
 		}
 		if err != storage.ErrFetching {

--- a/swarm/network/stream/delivery.go
+++ b/swarm/network/stream/delivery.go
@@ -209,10 +209,10 @@ R:
 
 		// this should be has locally
 		chunk, err := d.db.Get(req.Key)
-		if !bytes.Equal(chunk.Key, req.Key) {
-			panic(fmt.Errorf("processReceivedChunks: chunk key %s != req key %s (peer %s)", chunk.Key.Hex(), req.Key.Hex(), req.peer.ID()))
-		}
 		if err == nil {
+			if !bytes.Equal(chunk.Key, req.Key) {
+				panic(fmt.Errorf("processReceivedChunks: chunk key %s != req key %s (peer %s)", chunk.Key.Hex(), req.Key.Hex(), req.peer.ID()))
+			}
 			continue R
 		}
 		if err != storage.ErrFetching {

